conformance/src/type_checker.py

```diff
@@ -0,0 +1,213 @@
+"""
+Classes that abstract differences between type checkers.
+"""
+
+from abc import ABC, abstractmethod
+import json
+from pathlib import Path
+import re
+from subprocess import PIPE, run
+from typing import Sequence
+
+
+class TypeChecker(ABC):
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """
+        Returns the name of the type checker.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def install(self) -> None:
+        """
+        Ensures that the latest version of the type checker is installed.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_version(self) -> str:
+        """
+        Returns the current version string for the type checker.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def run_tests(self, test_files: Sequence[str]) -> dict[str, str]:
+        """
+        Runs the type checker on the specified test file and
+        returns the output.
+        """
+        raise NotImplementedError
+
+
+class MypyTypeChecker(TypeChecker):
+    @property
+    def name(self) -> str:
+        return "mypy"
+
+    def install(self) -> None:
+        run("pip install mypy --upgrade", shell=True)
```

Comment:
I find it's often more bulletproof to use `{sys.executable} -m pip`

Reply:
Thanks, I've made this change throughout.